tinygrad/runtime/cache_collector.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Optional, Dict, List, Callable, Tuple
+from weakref import ref
+from tinygrad.runtime.lib import RawBuffer
+
+class _CacheCollector:
```

Comment:
Line count is not important! This is really hard to read.

Reply:
added a bit of comments and make it cleaner in some places. lmk if this is better.